source/MaterialXTest/ShaderGen.cpp

```diff
@@ -868,3 +961,239 @@ TEST_CASE("LayeredSurface", "[shadergen]")
         file.close();
     }
 }
+
+TEST_CASE("Reference implementation validity", "[shadergen]")
```

Comment:
I don't have any assertions on this yet as it appears even the reference library is missing implementations. There some osl implementations missing and a number of glsl ones to come.